qubesbuilder/cli/cli_cleanup.py

```diff
@@ -0,0 +1,295 @@
+        component_dir = obj.config.artifacts_dir / "components" / component.name
+        if not component_dir.exists():
+            continue
+
+        versions = sorted(
+            [v for v in component_dir.iterdir() if v.name != "noversion"],
+            reverse=True,
+        )
+        for version in versions[keep_versions:]:
+            if obj.dry_run:
+                click.secho(f"DRY-RUN: {version}")
+            else:
+                click.secho(version)
+                shutil.rmtree(version)
+
+
+@click.command()
+@click.option(
+    "--log-retention-days", default=30, help="Number of days to keep logs."
+)
+@click.pass_obj
+def logs(obj: ContextObj, log_retention_days):
+    """
+    Cleanup old logs.
+    """
+    logs_dir = obj.config.logs_dir
+    cutoff_date = datetime.now() - timedelta(days=log_retention_days)
+
+    for log_file in logs_dir.iterdir():
+        if (
+            log_file.is_file()
+            and datetime.fromtimestamp(log_file.stat().st_mtime) < cutoff_date
+        ):
+            if obj.dry_run:
+                click.secho(f"DRY-RUN: {log_file}")
+            else:
+                click.secho(log_file)
+                log_file.unlink()
+
+
+@click.command()
+@click.option(
+    "--force",
+    default=False,
+    is_flag=True,
+    help="Force temporary directory cleanup.",
+)
+@click.pass_obj
+def tmp(obj: ContextObj, force: bool):
+    """
+    Cleanup temporary files and directories.
+    """
+    if not obj.config.temp_dir.exists():
+        return
+    for temp_dir in obj.config.temp_dir.iterdir():
+        # remove empty directory only unless force
+        if force or not list(temp_dir.iterdir()):
+            if obj.dry_run:
+                click.secho(f"DRY-RUN: {temp_dir}")
+            else:
+                click.secho(temp_dir)
+                shutil.rmtree(temp_dir)
+
+
+@click.command()
+@click.option(
+    "--all",
+    default=False,
+    is_flag=True,
+    help="Cleanup all caches including chroot and installer.",
+)
+@click.option(
+    "--chroot/--no-chroot",
+    default=False,
+    is_flag=True,
+    help="Cleanup chroot cache.",
+)
+@click.option(
+    "--installer/--no-installer",
+    default=False,
+    is_flag=True,
+    help="Cleanup installer cache.",
+)
+@click.option(
+    "--installer-chroot/--no-installer-chroot",
+    default=False,
+    is_flag=True,
+    help="Cleanup installer chroot cache.",
+)
+@click.option(
+    "--installer-templates/--no-installer-templates",
+    default=False,
+    is_flag=True,
+    help="Cleanup installer templates cache.",
+)
+@click.pass_obj
+def cache(
+    obj: ContextObj,
+    all: bool,
+    chroot: bool,
+    installer: bool,
+    installer_chroot: bool,
+    installer_templates: bool,
+):
+    """
+    Cleanup cache files and directories.
+    """
+    if all:
+        chroot = True
+        installer = True
+
+    to_delete = []
+    if chroot:
+        to_delete.append(obj.config.cache_dir / "chroot")
```

Comment:
There should be an option to remove chroots not relevant anymore (for no longer enabled dists)

Reply:
Right now it's going to remove all chroots, is it not what's wanted?

Reply:
I've added an option to only delete unused chroot based on distributions provided in builder.yml